### ISSUE TITLE
BatchNormalizationLayer useCntkEngine=true / false give different results

### ISSUE THREAD

**Issue by fwaris** [Microsoft/CNTK#1336]
My model trains fine using cuDNN batch normalization (useCntkEngine=false), however then I cannot use the model on CPU.

If I set useCntkEngine=true, the model trains but does not give me the same level of accuracy and does not perform as well on out-of-sample data. (using v2.0 beta 7). Note this takes longer, which is to be expected.

I am using the same SDG settings for both:
```
    SGD = [
        epochSize = 0 
        minibatchSize = 120
        learningRatesPerSample = 0.01*1000:0.001*1000:0.0005
        momentumPerMB = 0.9
        maxEpochs = 10000
        gradUpdateType = AdaGrad
...
]
```
Is the difference to be expected? Is there a way I can get the model to train better on the CPU version of batch normalization?

**Comment by sayanpa** [comment]
You can try Beta 9. Also CPU BN is supported in Beta 9.

**Comment by cha-zhang** [comment]
Indeed, since Beta 9, if you train with cuDNN, you may still use CPU to evaluate. 

The CNTK engine should be correct, as we have tested with networks such as ResNet and Inception. Are you sure the loss in accuracy is not caused by normal fluctuation due to lack of training examples?

**Comment by fwaris** [comment]
thanks I will try beta 9. I am not sure what causes the discrepancy between CPU and cuDNN implementations. The difference seems higher than normal variation and I have tested several times. 

I put together a script to live plot the CE and Err loss from the log file and the charts look appreciably different between the two types of batch normalization implementations. For anyone interested, the live chart script is here  http://www.fssnip.net/7Rn/title/Live-chart-of-CNTK-Error-Rate (consider it alpha code).

Also LayerNormalizationLayer allows me to train a smaller model and get similar results so I was using that in the meanwhile.
